senza/cli.py

```diff
@@ -462,12 +466,20 @@ def get_stack_refs(refs: list):
         else:
             try:
                 with open(ref) as fd:
-                    data = yaml.safe_load(fd)
+                    try:
+                        data = yaml.safe_load(fd)
+                    except yaml.YAMLError as e:
+                        raise InvalidDefinition(path=ref,
+                                                reason=str(e))
 
                 try:
                     ref = data['SenzaInfo']['StackName']
-                except (KeyError, TypeError):
-                    raise InvalidDefinition(path=ref)
+                except KeyError:
+                    raise InvalidDefinition(path=ref,
+                                            reason="SenzaInfo is missing")
```

Comment:
A better message would be "SenzaInfo section is missing or invalid", if the `StackName` is missing the same error message will be used.